### DISCUSSION TITLE
AI Generated Content

### DISCUSSION THREAD

**Discussion by ageddesi** [ageddesi/Mocked-API#195] [Ideas]
I would like to personally start exploring AI and as such I was thinking. Is there ways in which we can can use AI within this repo?

**Comment by devvspaces** [comment]
Yes, I have used python modules that automatically generate content

**Comment by devvspaces** [comment]
currently I see we are limited on data, we should have a way to automatically generate any kind on data we want based on the data type

**Reply by devvspaces** [reply]
For example, normal string, number, date, fullname, first name, last name

**Reply by devvspaces** [reply]
I would be like a module our developers will be able to use to generate json data, they would write  some thing like the data model and execute generate.

**Reply by devvspaces** [reply]
for example this is how a sample model would look like
```js
let Model = {
    name: {
        datatype: firstname,
    },
    receipt_id: {
        datatype: string,
        length: 40,
    },
    created: {
        datatype: timestamp,
    },
    quantity: {
        datatype: number,
    }
}
```

Then do something like `model.generate()`

**Reply by devvspaces** [reply]
This is just an idea, the `generate` would create rows of this model with random intuitive information

**Comment by ageddesi** [comment]
Yer we currently use faker.js for our data but would be nice to extend this as much as possible. 
Defining data models that auto generate would be ace.

**Reply by devvspaces** [reply]
Nice I would check out Faker and work on this
